rotkehlchen/globaldb/updates.py

```diff
@@ -301,7 +301,7 @@ def _parse_full_insert(self, insert_text: str) -> AssetData:
 
     def _apply_single_version_update(
             self,
-            cursor: DBCursor,
+            connection: DBConnection,
             version: int,
             text: str,
             conflicts: Optional[Dict[Asset, Literal['remote', 'local']]],
```

Comment:
Shouldn't this entire method use a savepoint also? So we have something like...
-> SAVEPOINT x
-----> SAVEPOINT x
-----> SAVEPOINT x

In this case, if the inner save goes through but for some reason, an exception occurs in the outer savepoint, the whole thing is rolled back.
What do you think?